src/stan_files/thurstonian_irt_predictor.stan

```diff
@@ -0,0 +1,159 @@
+functions {
+  /* cumulative log-PDF for a single response
+   * assumes the latent variable to be normal
+   * Args:
+   *   y: response category
+   *   mu: linear predictor
+   *   thres: ordinal thresholds
+   * Returns:
+   *   a scalar to be added to the log posterior
+   */
+   real cumulative_Phi_lpmf(int y, real mu, vector thres) {
+     int ncat = num_elements(thres) + 1;
+     real p;
+     if (y == 0) {
+       p = Phi(thres[1] - mu);
+     } else if (y == ncat - 1) {
+       p = 1 - Phi(thres[ncat - 1] - mu);
+     } else {
+       p = Phi(thres[y + 1] - mu) - Phi(thres[y] - mu);
+     }
+     return log(p);
+   }
+}
+data {
+  // model type
+  int<lower=1,upper=4> family;
+  // family == 1: bernoulli
+  // family == 2: cumulative
+  // family == 3: gaussian
+  // family == 4: beta
+  int<lower=1> N;  // total number of observations
+  // response variable
+  int Yint[family == 1 || family == 2 ? N : 0];
+  real Yreal[family == 3 || family == 4 ? N : 0];
+  int<lower=1> N_item;
+  int<lower=1> N_itemC;  // item pairs
+  int<lower=1> N_person;
+  int<lower=1> N_trait;
+  int<lower=0> N_item_fix;
+  int<lower=0> N_item_est;
+  // indices over N
+  int<lower=1> J_item1[N];
+  int<lower=1> J_item2[N];
+  int<lower=1> J_itemC[N];
+  int<lower=1> J_person[N];
+  int<lower=1> J_trait1[N];
+  int<lower=1> J_trait2[N];
+  int<lower=1> J_item_fix[N_item_fix];
+  int<lower=1> J_item_est[N_item_est];
+  // indicate inverted items
+  int<lower=0> N_item_pos;
+  int<lower=0> N_item_neg;
+  int<lower=1> J_item_pos[N_item_pos];
+  int<lower=1> J_item_neg[N_item_neg];
+  // indicate items used in multiple blocks
+  int<lower=0> N_item_equal;
+  int<lower=1> J_item_equal[N_item_equal];
+  int<lower=1> J_item_orig[N_item_equal];
+  // number of response categories in ordinal models
+  // should be set to 2 for other families
+  int<lower=2> ncat;
+}
+transformed data {
+  vector<lower=0>[N_item_fix] psi_fix;  // fixed item SDs
+  psi_fix = rep_vector(1.0, N_item_fix);
+}
+parameters {
```

Comment:
What did you change from the original model fitting stan file and the predictor file? I am a little confused as to how many parameters are included in the model here. Wouldn't the only new parameters be the new trait scores?